cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -412,16 +441,33 @@ mod tests {
         assert!(actual.path.is_absolute());
     }
 
-    /// Write a manifest file with invalid toml to ensure we can catch
     #[test]
     fn cache_activation_path() {
         let (flox, temp_dir) = flox_instance();
 
         let environment_temp_dir = tempfile::tempdir_in(&temp_dir).unwrap();
         let pointer = PathPointer::new("test".parse().unwrap());
 
-        let mut env =
-            PathEnvironment::init(pointer, environment_temp_dir.path(), temp_dir.path()).unwrap();
+        let mut env = PathEnvironment::init(
+            pointer,
+            environment_temp_dir.path(),
+            temp_dir.path(),
+            &flox.system,
+        )
+        .unwrap();
+        // building requires packages in the manifest, install one so the build succeeds and the outlink is built
+        let pkgs = vec![
+            PackageToInstall::from_str("hello").expect("couldn't create package from 'hello'")
+        ];
+        env.install(&pkgs, &flox).expect("couldn't install 'hello'");
+        // Need to touch the manifest so that its mtime is more recent than the outlink
+        Command::new("touch")
+            .arg(
+                env.manifest_path(&flox)
+                    .expect("couldn't get manifest path"),
+            )
+            .output()
+            .expect("couldn't touch manifest");
```

Comment:
That's not what this is doing. The assertion checks whether we need a rebuild. In the original test the manifest exists but we don't have an outlink, so it needs to be built. Here, since this test is failing in CI with "no packages for <system>" I tried installing packages as a baseline, then touch the manifest so that `needs_rebuild` returns `true` like it did in the original test.

Reply:
the first test was meant to catch "needs to rebuild because we have not build yet".
I don't think we should throw any error if there is no package though, that seems like a regression..